### PR TITLE
(#3242) Implement ncids breadcrumbs

### LINKED ISSUE #3242: Enabler: Build Drupal Block for NCIDS Breadcrumb
# Build Drupal Block for NCIDS Breadcrumb
> **ESTIMATE** 3

>_**NOTE:**_ We are able to use the existing Drupal back-end block for the breadcrumbs (see below). The only things that are outstanding on this ticket are:
>  * implementing the analytics
>  * 2 bugs where the page title portion of the breadcrumbs is breaking
>     * #3419
>       * #3366
>  * adding some way to support the current breadcrumbs on current templates between deployment of code and update of the web site to use the new theme. (This may only be an ODE issue right now)
> * The "Show" or "Hide" breadcrumb option needs to stay the same. We will update all of the microsites to have "Show" when launched.

The NCIDS breadcrumb markup differs from the current [CGDP Breadcrumbs](https://github.com/NCIOCPL/cgov-digital-platform/blob/master/docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/Plugin/Block/Breadcrumb.php).  We will be reusing the existing Breadcrumb block, but will update it to display the ?? title after the parent sections. This will modify the DCEG Connect site.

This should:
1. Use the wrapping variant
2. For the parent navigation items in the breadcrumbs, display the Navigation Label if it exists, otherwise display the Site Section Name.
3. For the current page you are on, display the page title.

### Acceptance Criteria
```gherkin
Scenario: User is on a c.gov site on desktop, viewing a L1 home and landing page
  Given they are on a L1 that uses the home and landing page content type (e.g. About Cancer)
  Then they will not see the breadcrumbs
```

```gherkin
Scenario: User is on a c.gov site on desktop, viewing a non-home-and-landing-page L1
  Given they are on an L1 that is any content type other than home and landing (e.g. Publications)
  Then they will see the breadcrumbs
    And the breadcrumbs will display {Root Site Section} > {Current Page Title}
    And {Root Site Section} will be a link to the landing page of the root site section
```
*Note*: The root site section name is for the language that you are viewing.

```gherkin
Scenario: User is on cancer.gov on desktop, viewing a L2
  Given they are on a L2 or lower (e.g. Adjusting to Cancer)
  Then they will see all breadcrumbs
    And the breadcrumbs will display starting with {Root Site Section}
    And {Root Site Section} will be a link to the landing page of the root site section
    And the breadcrumbs will display ending with the page they are on 
    And each level will be separated with a caret
    And each level will be a link to the landing page of the section
    But the page they are on will not be hyperlinked
```

```gherkin
Scenario: User is on a c.gov site on a mobile device
  Given they are on a c.gov site on a mobile device >480
  Then they will only see the parent of the page they are on
  And it will be hyperlinked
  And there will be an arrow to the left of it
```

### Acceptance Criteria for Analytics
``` gherkin
Given a user is on any page of cancer.gov other than home and landing pages
  And the user clicks on a link within the breadcrumbs section of the page
When the user clicks on a link in the mega menu
Then there should be an NCIDataLayer Other event 
  And the event has the name "Breadcrumbs:LinkClick"
  And the event has the the linkname "Breadcrumbs:LinkClick"
  And this event should have the following data:
      | key      | value                                                                                  |
      | linkText | Text of the link clicked                                                               |
      | location | Breadcrumbs                                                                            |
      | section  | The value of the hierarchy of the link clicked. ie:                                    |
      |          | If the top-level link is clicked, return L1. If the second link is clicked, return L2. |
```

--------

### Prerequisites
- [ ] https://github.com/NCIOCPL/ncids/issues/230
- [ ] #3241

### LINKED ISSUE #3419: The token current-page:title double encodes html entities
In reviewing the inner pages implementation we noticed a page with the title `TCGA&#039;s Study of Lung Squamous Cell Carcinoma` is displaying the HTML entity, visually, in the breadcrumbs. (Which is happening because it is double encoding the entities.) The URL you can see this is `/about-nci/organization/ccg/research/structural-genomics/tcga/studied-cancers/lung-squamous`, but that is from production. So as a developer, you will probably need to create add a node through YAML content to see it on a local machine. 

Notes:
Spanish characters seem to work: https://ncigovcdode476.prod.acquia-sites.com/espanol/investigacion/areas/biologia

Added quotes to this page title and they are showing up with HTML entity in the breadcrumb: https://ncigovcdode476.prod.acquia-sites.com/about-nci/organization/ccg/research/structural-genomics/tcga/studied-cancers/cholangiocarcinoma

Also tried entering the apostrophe directly in the page title field and from pasting in and both result in the HTML showing in the breadcrumb

### Additional details / screenshot
image

### Related Tickets
* #3242

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #3242 
Closes #3419

ODE https://ncigovcdode495.prod.acquia-sites.com/

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-breadcrumb/__tests__/usa-breadcrumb.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		// Hack to clean out the dom.
+		document.getElementsByTagName('body')[0].innerHTML = '';
+		jest.resetAllMocks();
+	});
+
+	it('sends analytics when the same item is clicked', async () => {
+		// Lets make a spy to ensure that trackOther is called correctly
+		const spy = jest.spyOn(eddlUtil, 'trackOther');
+
+		// Inject the HTML into the dom.
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-breadcrumb/__tests__/usa-breadcrumb.test.ts:22]: Clean please

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
One item.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-breadcrumb/__tests__/usa-breadcrumb.test.ts (hunk 1 above)
Comment (bennettcc) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-breadcrumb/__tests__/usa-breadcrumb.test.ts:22]: Tagging @olitharp-nci

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
The flag for always showing the breadcrumbs, even when home is the only item, needs to be set to show. In the new world, there is never a condition where the breadcrumb "Home" shows. It is always going to be "Home > current page". You can see this on the `/pets` -- the bread crumb should say "Home > Pets".

This will be updating `docroot/profiles/custom/cgov_site/config/install/block.block.ncids_trans_ncids_breadcrumb.yml` 

```
settings:
  id: ncids_breadcrumb
  label: 'NCIDS Breadcrumb'
  provider: cgov_core
  label_display: '0'
```

Will become
```
settings:
  id: ncids_breadcrumb
  label: 'NCIDS Breadcrumb'
  provider: cgov_core
  label_display: '1'
```

**Comment by bennettcc** [comment]
@welshja and @nicolecrom confirmed that analytics AC's look good on ODE496

**Comment by olitharp-nci** [comment]
> The flag for always showing the breadcrumbs, even when home is the only item, needs to be set to show. In the new world, there is never a condition where the breadcrumb "Home" shows. It is always going to be "Home > current page". You can see this on the `/pets` -- the bread crumb should say "Home > Pets".
> 
> This will be updating `docroot/profiles/custom/cgov_site/config/install/block.block.ncids_trans_ncids_breadcrumb.yml`
> 
> ```
> settings:
>   id: ncids_breadcrumb
>   label: 'NCIDS Breadcrumb'
>   provider: cgov_core
>   label_display: '0'
> ```
> 
> Will become
> 
> ```
> settings:
>   id: ncids_breadcrumb
>   label: 'NCIDS Breadcrumb'
>   provider: cgov_core
>   label_display: '1'
> ```


Updates needed (for future reference) --

```yml
# block.block.ncids_trans_ncids_breadcrumb.yml

settings:
  id: ncids_breadcrumb
  label: 'NCIDS Breadcrumb'
  provider: cgov_core
  label_display: '0'
  show_home_alone: show
```

```yml
# cgov_core.schema.yml

# Settings for the NCIDS Breadcrumb
block.settings.cgov_breadcrumb:
  type: block_settings
  label: 'NCIDS Breadcrumb block configuration'
  mapping:
    show_home_alone:
      type: string
      label: 'Root Site Section Display'
```

https://ncigovcdode495.prod.acquia-sites.com/pets

**Review by bryanpizzillo — APPROVED** [review]
Looks great!

**Pull request merged**